### PR TITLE
tools: Use 'make-rpms' to build host native packages

### PR DESCRIPTION
**Pull request by stefwalter**
This is the quick path to testing out a Cockpit commit on an RPM
based system. It doesn't require mock, or a VM or anything else
besides all the development dependencies.

### CODE REVIEW

File: tools/make-rpms [hunk 1]
```diff
@@ -16,39 +16,23 @@
...
-        # has been cleaned previously.
-		mock_clean_opts="--no-cleanup-after"
-		;;
 	-v|--verbose)
-		mock_opts="$mock_opts --verbose"
-		;;
-	-q|--quick)
-		mock_define="selinux" mock_value="0"
-		mock_opts="$mock_opts --offline --nocheck"
+        opts=""
```
Comment (mvollmer) [inline tools/make-rpms]: Indentation

File: tools/make-rpms [hunk 2]
```diff
@@ -16,39 +16,23 @@
...
 }
 
-mock_opts=""
-mock_clean_opts="--no-clean --no-cleanup-after"
-mock_define="unused"
-mock_value="0"
+opts="--quiet"
 
-args=$(getopt -o "h,v,c,q,r:" -l "help,verbose,clean,quick" -- "$@")
+args=$(getopt -o "h,v,q,r:" -l "help,verbose,quick" -- "$@")
```
Comment (mvollmer) [inline tools/make-rpms]: --quick is not implemented.  I think we should.

Comment (stefwalter) [inline tools/make-rpms]: Removed.

File: tools/make-rpms (hunk 1 above)
Comment (stefwalter) [inline tools/make-rpms]: Fixed.

File: tools/make-rpms [hunk 3]
```diff
@@ -67,82 +51,17 @@ if [ $# -ne 0 ]; then
...
-cp --preserve=timestamps /etc/mock/logging.ini $mockcfgdir
-mockcfg=$mockcfgdir/$os-$arch-cockpit.cfg
+tmpdir=$(mktemp -d)
+rpmbuild --rebuild $opts \
+    --define "_sourcedir `pwd`" \
+    --define "_specdir `pwd`" \
+    --define "_builddir `pwd`" \
+    --define "_srcrpmdir `pwd`" \
+    --define "_rpmdir $tmpdir" \
+    --define "_buildrootdir `pwd`/.build" \
```
Comment (dperpeet) [inline tools/make-rpms]: this doesn't get deleted after the build

**Comment by stefwalter** [comment]
Changed to do all building inside of a temporary directory, which gets removed afterwards. Also added back the ```--quick``` option, which skips checks.

File: tools/cockpit.spec [hunk 4]
```diff
@@ -142,13 +142,15 @@ The Cockpit Web Service listens on the network, and authenticates users.
 %setup -q
 
 %build
+exec 2>&1
 %if %{defined gitcommit}
 env NOCONFIGURE=1 ./autogen.sh
 %endif
-%configure --disable-static --disable-silent-rules --with-cockpit-user=cockpit-ws --with-branding=%{branding} --with-selinux-config-type=etc_t
+%configure --disable-silent-rules --with-cockpit-user=cockpit-ws --with-branding=%{branding} --with-selinux-config-type=etc_t
```
Comment (mvollmer) [inline tools/cockpit.spec]: This looks like an unrelated change. Why is --disable-static removed?

Comment (stefwalter) [inline tools/cockpit.spec]: Because it's not supported, and just makes configure.ac spew noise. The commit this is in is one that removes noise.

Comment (mvollmer) [inline tools/cockpit.spec]: Thanks, makes sense.
